src/agent.rs

```diff
@@ -1,17 +1,17 @@
-use ssh_agent_client_rs::{Client, Result};
-use ssh_key::{PublicKey, Signature};
+use ssh_agent_client_rs::{Client, Identity, Result};
+use ssh_key::Signature;
 
 /// A small trait defining the two methods of ssh_agent_client_rs::Client to simplify testing
 pub trait SSHAgent {
-    fn list_identities(&mut self) -> Result<Vec<PublicKey>>;
-    fn sign(&mut self, key: &PublicKey, data: &[u8]) -> Result<Signature>;
+    fn list_identities(&mut self) -> Result<Vec<Identity>>;
+    fn sign(&mut self, identity: &Identity, data: &[u8]) -> Result<Signature>;
```

Comment:
This should match the upstream sign() signature: `sign(&mut self, key: impl Into<Identity<'a>>, data: &[u8])`.

That way you can pass in, keys certs and identities as you please